src/use_case/delete_exercise/DeleteExerciseInteractor.java

```diff
@@ -1,5 +1,41 @@
 package use_case.delete_exercise;
 
-public class DeleteExerciseInteractor {
+import use_case.add_exercise.AddExerciseDataAccessInterface;
+import use_case.add_exercise.AddExerciseInputData;
+import use_case.add_exercise.AddExerciseOutputBoundary;
 
+public class DeleteExerciseInteractor implements DeleteExerciseInputBoundary{
+
+    /**
+     * @param deleteExerciseInputData
+     */
+    final DeleteExerciseDataAccessInterface deleteExerciseDataAccessObject;
+
+    final DeleteExerciseOutputBoundary deleteExercisePresenter;
+
+    public DeleteExerciseInteractor(DeleteExerciseDataAccessInterface deleteExerciseDataAccessObject,
+                                    DeleteExerciseOutputBoundary deleteExerciseOutputBoundary) {
+        this.deleteExerciseDataAccessObject = deleteExerciseDataAccessObject;
+        this.deleteExercisePresenter = deleteExerciseOutputBoundary;
+    }
+
+    @Override
+    public void execute(DeleteExerciseInputData deleteExerciseInputData) {
+        int id = deleteExerciseInputData.getId();
+        String exercise_name = deleteExerciseInputData.getName();
+
+        // Check routine with id corresponding to id exists
+        if (!deleteExerciseDataAccessObject.existsById(id)) {
+            // Use exceptions? instead
+            // addExercisePresenter.prepareFailView("Routine" + id + "does not exist"); - temp until presenter implemented
+            // Check exercise with name corresponding to exercise_name exists
+        } else if (!deleteExerciseDataAccessObject.existsByName(exercise_name)) {
+
+        } else {
+            // Update the routine corresponding to id by adding the exercise corresponding to exercise_name
+            deleteExerciseDataAccessObject.updateRoutine(id, exercise_name);
```

Comment:
Change the method name to something like deleteExercise to be more descriptive?

Reply:
> also assumes that you dont have two of the same exercise in a routine

I know, this just deletes the first one. I feel like it would easier for someone who has multiple of the same exercises in the routine to delete them one at a time vs. someone who has multiple but wants to delete only one to have to add the others manually, if that makes sense. If I have time, I was thinking about an implementation with an individual "delete" button for each exercise in the routine.

Reply:
i understand this, i think it'll just have to be an assumption we make unless we want to add indexing: that would require an index variable in the input data (when the user selects an exercise, it will send to your controller also which index this exercise is, which then maybe you don't need exercise name at all? I'm rambling) which to be honest i think it's doable (it's a corner case we should at least acknowledge) but i totally understand that it's kinda complicated haha